Fix robot windows of robotbenchmark simulations

BenjaminDeleze opened the pull request:
**Description**
Restore the robot windows of the robotbenchmark simulations that were broken with the removal of JQuery:

- [x] Change JQuery expression with vanilla JS
- [x] Change `var` to `let`/`const`


Fixes #3182

stefaniapedrazzi reviewed (changes requested):
Thank you. Globally seems fine.
I started testing the functionality, I will then check the code changes.

I just found some issues with images and the width of the robot window content.
In the `humanoid_sprint.wbt` and `wall_following.wbt` robot windows, the content width is not correctly updated when increasing the width of the robot window widget and images are partially outside the content area.

Other issues were already reproducible in the master branch.

stefaniapedrazzi commented on projects/samples/robotbenchmark/include/tabs.css:
```diff
@@ -0,0 +1,64 @@
+.radio{
+  display:none;
+}
+#one:checked ~ .panels #one-panel,
+#two:checked ~ .panels #two-panel,
+#three:checked ~ .panels #three-panel,
+#four:checked ~ .panels #four-panel{
+  display:block
+}
```
Is not possible to use a class instead of ids?
That would be more generic and scalable.

stefaniapedrazzi commented on projects/samples/robotbenchmark/include/tabs.css:
```diff
@@ -0,0 +1,64 @@
+#four:checked ~ .panels #four-panel{
+  display:block
+}
+#one:checked ~ .tabs #one-tab,
+#two:checked ~ .tabs #two-tab,
+#three:checked ~ .tabs #three-tab,
+#four:checked ~ .tabs #four-tab{
```
Is not possible to use classes selectors instead of ids?
That would be more generic and scalable.

stefaniapedrazzi commented on projects/samples/robotbenchmark/highway_driving/plugins/robot_windows/highway_driving/highway_driving.html:
```diff
@@ -2,57 +2,62 @@
-          This benchmark aims at driving the farther as possible in one minute without colliding any other vehicles or static objects.
-        </p>
+    <div id='infotabs' class='wrapper webotsTabs webotsRobotWindowTabs'>
+      <input class="radio" id="one" name="group" type="radio" checked>
+      <input class="radio" id="two" name="group" type="radio">
+      <input class="radio" id="three" name="group" type="radio">
+      <div class="tabs">
+        <label class="tab" id="one-tab" for="one">Overview</label>
+        <label class="tab" id="two-tab" for="two">Metrics</label>
+        <label class="tab" id="three-tab" for="three">Instructions</label>
```
I would rather keep specific ids describing the content (overview, metric, instructions, etc.) instead of using generic "one", "two", "three", etc. that doesn't tell anything about the content.

Same for panel ids, etc.

stefaniapedrazzi commented on projects/samples/robotbenchmark/viewpoint_control/plugins/robot_windows/viewpoint_control/viewpoint_control.html:
```diff
@@ -2,45 +2,52 @@
+          <p>The tab entitled <a class='tablink' onclick='$("#infotabs").tabs({active:1});'>Metrics</a> measures your progress in an exercise where
+            you have to place the viewpoint at a specific position and orientation. It also allows you to record your performance in
+            the robotbenchmark database (if you are logged-in).</p>
+          <p>The tab entitled <a class='tablink' onclick='$("#infotabs").tabs({active:2});'>Instructions</a> explains you how to use the mouse to
```
These links to other tabs in the same document are no longer working.
They should be fixed or removed.

Same on  line 21 an 39.

stefaniapedrazzi reviewed (commented):
In `viewpoint_control.js` there is still a reference to the THREE library.
The code to evaluate the viewpoint position is currently only working when running it from the web interface and it doesn't seem we are interested in further supporting this benchmark.
But it would be good to clean it up and remove all the references to three.js.

BenjaminDeleze commented:
I think I addressed all the comments
  -  To use classes instead of ids was a bit tricky and I could not find another solution than introducing a bit of JS.
  -  I removed the viewpoint control benchmark.

stefaniapedrazzi commented:
I'm reviewing the changes now.
Could you please add an entry in the ChangeLog about the removal of the viewpoint_control benchmark?

stefaniapedrazzi reviewed (changes requested):
Thank you!

Images size is now updated correctly on window resize.
But even at maximum size, the legend of the plot in `humanoid_sprint.wbt` is unreadable.
Maybe it is possible to increase the maximum size of the robot window content.

BenjaminDeleze commented:
I totally removed the max-width, I do not think it is useful.
However, it was not sufficient as the legend of the plot is not readable either in robotbenchmark.net, so I also removed the predefined size of the image and it seems to display well now.

stefaniapedrazzi commented on projects/samples/robotbenchmark/highway_driving/plugins/robot_windows/highway_driving/highway_driving.html:
```diff
@@ -2,59 +2,79 @@
       </div>
     </div>
     <script src="highway_driving.js"></script>
+    <script>
+      let height;
+      function adaptPositionOfText() {
+        let tabs = document.getElementsByClassName('tab')
+        if(!tabs.length > 0)
+          return
+        let newHeight = tabs[tabs.length - 1].getBoundingClientRect().top + tabs[tabs.length - 1].getBoundingClientRect().height;
+        if(newHeight === height)
+          return;
+        height = newHeight;
+        let panels = document.getElementsByClassName('panel');
+        for (panel of panels) {
+          panel.style.top = height + 'px';
+        };
+      }
+      window.addEventListener('resize', adaptPositionOfText);
+      adaptPositionOfText();
+    </script>
```
Just a question: why this was not needed before when using ids instead of classes in the CSS?
Shouldn't work exactly the same?

BenjaminDeleze replied:
Because with ids we had a structure like:
```
<input></input>
<input></input>
<input></input>

<label class=tabs></label>
<label class=tabs></label>
<label class=tabs></label>

<div class=panel></panel>
<div class=panel></panel>
<div class=panel></panel>
```
Because we were using IDs, it was easy to connect one input to one particular div.
And also the element were displayed in order: the tabs on top, the panel below.

However when switching for classes the structures changed like this:
```
<input></input>
<label class=tabs></label>
<div class=panel></panel>

<input></input>
<label class=tabs></label>
<div class=panel></panel>

<input></input>
<label class=tabs></label>
<div class=panel></panel>
```

It is the case because as we don't have IDs anymore, it is way harder to connect one specific input to a specific panel. The simplest solution with only CSS I found was to connect the input to the nearest panel below it. But as you can see the structure is not the same without further changes it would display like:

```
<Overview>
text_of_panel, text_of_panel, text_of_panel, text_of_panel, text_of_panel, text_of_panel
text_of_panel, text_of_panel, text_of_panel, text_of_panel, text_of_panel, text_of_panel, 
text_of_panel, text_of_panel, text_of_panel, text_of_panel, text_of_panel, text_of_panel
<Metric>
<Instructions>
```

To solved that I made the position of the panels absolute. Problem: if we resize the window and the buttons are placed on several columns, we need to update the position of the panels to always be just below the lowest button. And that what the JS code does. 

I am not sure this is an optimal solution, but it is the simplest I could come up with without IDs and then scalable.

BenjaminDeleze commented:
Can you approved it again please?
I had to fix a conflict in the changelog.

stefaniapedrazzi commented on docs/reference/changelog-r2021.md:
```diff
@@ -129,6 +129,8 @@ Released on June, Xth, 2021.
     - Deleted deprecated DifferentialWheels node ([#2749](https://github.com/cyberbotics/webots/pull/2749)).
     - Changed structure of the [projects/samples/howto]({{ url.github_tree }}/projects/samples/howto) directory, so each demonstration is in a dedicated directory ([#2639](https://github.com/cyberbotics/webots/pull/2639)).
+    - Remove the viewpoint_control benchmark that was not working in Webots ([#3192](https://github.com/cyberbotics/webots/pull/3192)).
+
```
```suggestion
```

stefaniapedrazzi reviewed (changes requested):
Yes, could you just please remove this empty line?

stefaniapedrazzi reviewed (approved):
Thank you
